deepmd/pt/model/descriptor/descriptor.py

```diff
@@ -206,6 +208,32 @@ class DescriptorBlock(torch.nn.Module, ABC):
     __plugins = Plugin()
     local_cluster = False
 
+    def __init__(
+        self,
+        ntypes: int,
+        exclude_types: List[Tuple[int, int]] = [],
+    ):
+        super().__init__()
+        _exclude_types: Set[Tuple[int, int]] = set()
+        for tt in exclude_types:
+            assert len(tt) == 2
+            _exclude_types.add((tt[0], tt[1]))
+            _exclude_types.add((tt[1], tt[0]))
+        # ntypes + 1 for nlist masks
+        self.type_mask = np.array(
+            [
+                [
+                    1 if (tt_i, tt_j) not in _exclude_types else 0
+                    for tt_i in range(ntypes + 1)
+                ]
+                for tt_j in range(ntypes + 1)
+            ],
+            dtype=np.int32,
+        )
+        # (ntypes+1 x ntypes+1)
+        self.type_mask = torch.from_numpy(self.type_mask).view([-1])
```

Comment:
Should `self.type_mask` be saved through `register_buffer`, so it can be saved into `model.pt`? Or we allow user to change `exclude_types` each time when restart from saved checkpoint, even if it doesn't match each other?

Reply:
Additionally, `device` missed here for `torch.from_numpy`, GPU may fail.